### DISCUSSION TITLE
Error in browser console (target_arch = "wasm32")

### DISCUSSION THREAD

**Discussion by azecm** [slint-ui/slint#6779] [General]
How to remove/resolve the error that winint gives
- Uncaught Error: Using exceptions for control flow, don't mind me. This isn't actually an error!
![Снимок экрана 2024-11-13 в 20 10 31](https://github.com/user-attachments/assets/c0275f57-ea5a-4939-9cf4-a969c0fad5b6)

**Comment by ogoffart** [comment]
I believe we need to wrap the call to `init()` in a try{}catch{} block to silent this.

**Reply by azecm** [reply]
This is an obvious option

I was hoping for a more elegant solution

Thank you

**Comment by azecm** [comment]
checked this option (try{}catch{})

- doesn't help
since init is asynchronous

Of course, I would like to indicate some additional parameters at startup

to avoid this error

**Comment by azecm** [comment]
the code below will work only the beginning

```rust
#[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(start))]
pub fn main_wasm() {
    console_error_panic_hook::set_once();
    wasm_logger::init(wasm_logger::Config::default());
    log::info!("init");
    if let Ok(main_window) = HelloWorld::new() {
        log::info!("init-1");
        match main_window.run() {
            Ok(_)=>{
                log::info!("init-2");
            }
            Err(error)=>{
                log::info!("init-3");
                log::error!("failed: {:?}", error);
            }
        }
    }
    log::info!("started");
}
``` 

![Снимок экрана 2024-11-13 в 20 52 23](https://github.com/user-attachments/assets/1e1bc7d8-b6ba-4415-8860-eeca9775faa2)

**Comment by azecm** [comment]
intercepted it like that

```javascript
    try {
        await init();
    }
    catch (e){}
```
